pandas/tests/io/parser/mangle_dupes.py

```diff
@@ -40,3 +40,25 @@ def test_thorough_mangle(self):
         df = self.read_csv(StringIO(data), sep=",", mangle_dupe_cols=True)
         assert list(df.columns) == ["a", "a.1", "a.3", "a.1.1",
                                     "a.2", "a.2.1", "a.3.1"]
+
+    def test_thorough_mangle_names(self):
+        # see gh-17095
+        data = "a,b,b\n1,2,3"
+        names = ["a.1", "a.1", "a.1.1"]
+        df = self.read_csv(StringIO(data), sep=",", names=names,
+                           mangle_dupe_cols=True)
+        assert list(df.columns) == ["a.1", "a.1.1", "a.1.1.1"]
```

Comment:
I would aim for a deprecation cycle.  `_maybe_dedup_names` has been well-ingrained for sometime.  In addition, the function is actually used to dedup column names when the data portion of the CSV is empty.  Thus, I would like to give some more thought on how best to deprecate deduping `names` without issuing a warning in the empty case.